src/olympia/files/utils.py

```diff
@@ -554,7 +560,42 @@ def copy_over(source, dest):
     shutil.rmtree(source)
 
 
-def extract_xpi(xpi, path, expand=False):
+def get_all_files(folder, strip_prefix='', prefix=None):
+    """Return all files in a file/directory tree.
+
+    :param folder: The folder of which to return the file-tree.
+    :param strip_prefix str: A string to strip in case we're adding a custom
+                             `prefix` Doesn't have any implications if
+                             `prefix` isn't given.
+    :param prefix: A custom prefix to add to all files and folders.
+    """
+
+    all_files = []
+
+    # Not using os.path.walk so we get just the right order.
```

Comment:
That was actually copied from the method from `FileViewer` and I didn't want to change this. You probably can, I don't know what grantees we currently have, it seems to me that we currently rely on the file-list we get from the zip-file or something?